domain/schema/model/sql/0015-charm.sql

```diff
@@ -67,9 +67,32 @@ CREATE TABLE charm (
     CHECK (source_id = 0 OR source_id = 1 AND architecture_id >= 0)
 );
 
+-- This ensures that the reference name and revision are unique. This is to
+-- ensure that we don't have two charms with the same reference name and
+-- revision. If this happens, we can just link the application to the existing
+-- charm.
 CREATE UNIQUE INDEX idx_charm_reference_name_revision
-ON charm (reference_name, revision);
+ON charm (source_id, reference_name, revision);
 
+-- The charm_local_sequence table is used to keep track of the sequence number
+-- of the local charm. This is used to determine the revision of the charm.
+-- For that reason we can not use a revision directly in the table.
+--
+-- Charms are unique rows in the database and the metadata is immutable. To
+-- locate the charm, we can not use the charm.uuid, as this doesn't allow us
+-- to group them. Instead, the group is the source_id and the reference_name.
+-- This is the unique identifier for the charm.
+CREATE TABLE charm_local_sequence (
```

Comment:
Can we put "sequence" as the prefix for all such tables? That will make them groupable and obvious.